### PR TITLE
Restructure lib

### PR DESCRIPTION
**Pull request by andy-isoc**
Add ISO 2-letter codes for regions and sub-regions

Ugh, sorry the diff is really noisy on the main lib code here. All I did was remove the `Regions` class and make everything a first-class function, and then add the region ISO 2-letter codes.

### CODE REVIEW

**Comment by andy-isoc** [comment]
There are two 2-letter codes used by `heart` that aren't included here:
- XA: the World
- XG: "unclassified"

I'm in two minds about whether we should include them here or not. I can't think of a obvious way to include them and it is (currently) only one place in heart that needs them (DNSSEC validation) so I'm erring towards leaving them out.

File: django_countries_hdx/__init__.py [hunk 1]
```diff
@@ -7,261 +7,281 @@
...
-                            HDX.get_iso2_from_iso3(code),
-                            HDX.get_country_name_from_iso3(code)
-                        )
-                        for code in HDX.get_countries_in_region(region_code)
-                    ],
-                    key=lambda x: x[1]  # Sort by country name
-                )
-            }
-            for region_code in sorted(region_codes, key=lambda x: self.get_region_name(x))
+    :param regions: Boolean to determine whether to use regions or subregions.
```
Comment (madofo) [inline django_countries_hdx/__init__.py:171]: nit: no params?

File: django_countries_hdx/__init__.py [hunk 2]
```diff
@@ -7,261 +7,281 @@
...
-                        for code in HDX.get_countries_in_region(region_code)
-                    ],
-                    key=lambda x: x[1]  # Sort by country name
-                )
-            }
-            for region_code in sorted(subregion_codes, key=lambda x: self.get_region_name(x))
+def get_countries_by_subregion() -> dict[int, dict[str, list[tuple[str, str]]]]:
+    """Retrieves lists of countries keyed by region, with region name and country tuples.
+
+    :param regions: Boolean to determine whether to use regions or subregions.
```
Comment (madofo) [inline django_countries_hdx/__init__.py:204]: nit: no params?

**Review by madofo — APPROVED** [review]
I suppose we just have to handle XA as a special case whenever we want to use that data? 'undefined' seems less problematic.

File: django_countries_hdx/__init__.py (hunk 1 above)
Comment (andy-isoc) [reply django_countries_hdx/__init__.py]: I've added better typing for the return value as well.

File: django_countries_hdx/__init__.py (hunk 2 above)
Comment (andy-isoc) [reply django_countries_hdx/__init__.py]: Also better return typing as above.

**Comment by andy-isoc** [comment]
I don't think Python has proper constants otherwise I would add those codes to the lib as constants. We could provide helper functions for them I guess. But not sure it's worth it. If we start to use them in more places then we can take another look. We usually use the M49 region and sub-regions codes I think.

**Review by garrettc — APPROVED** [review]

**Pull request merged**